apps/condo/domains/billing/access/RegisterBillingReceiptFileService.js

```diff
@@ -14,7 +15,14 @@ async function canRegisterBillingReceiptFile ({ authentication: { item: user },
         if (!context) {
             return false
         }
-        return await checkBillingIntegrationsAccessRights(user.id, [context.integration])
+        const hasBillingAccess = await checkBillingIntegrationsAccessRights(user.id, [context.integration])
+        if (hasBillingAccess) {
+            return true
+        }
+        const hasB2BAccess = await checkB2BAccessRightsToBillingContext(args, context)
+        if (hasB2BAccess) {
+            return true
+        }
```

Comment:
Сan there be a situation when there is access to the b2b application, but no access to billing? Or vice versa?
Does it make sense to change the order to speed up the request (by the popularity of the connection method)?

Reply:
the main type for service user will be through checkBillingIntegrationsAccessRights
only if it's failed it will check checkB2BAccessRightsToBillingContext